From 6a1cfe9a55253b2913e8d58b5b56df941959a741 Mon Sep 17 00:00:00 2001
From: Sven van Haastregt <sven.vanhaastregt@arm.com>
Date: Wed, 20 Apr 2022 09:51:43 +0100
Subject: [PATCH 030/172] .clang-tidy: allow recursion

The code already uses recursion in various places and this check fails
whenever a change is made in the recursion chain.

(cherry picked from commit 28fdb7ace62ac33af43374f8e52f6479073ae594)
---
 .clang-tidy | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/.clang-tidy b/.clang-tidy
index 698e7e10..f8f5c626 100644
--- a/.clang-tidy
+++ b/.clang-tidy
@@ -1,5 +1,5 @@
-Checks: '-*,clang-diagnostic-*,llvm-*,misc-*,-misc-unused-parameters,readability-identifier-naming,-llvm-header-guard'
-WarningsAsErrors: 'llvm-*,misc-*,-misc-unused-parameters,readability-identifier-naming,-llvm-header-guard'
+Checks: '-*,clang-diagnostic-*,llvm-*,misc-*,-misc-no-recursion,-misc-unused-parameters,readability-identifier-naming,-llvm-header-guard'
+WarningsAsErrors: 'llvm-*,misc-*,-misc-no-recursion,-misc-unused-parameters,readability-identifier-naming,-llvm-header-guard'
 CheckOptions:
   - key:             readability-identifier-naming.ClassCase
     value:           CamelCase
@@ -17,4 +17,3 @@ CheckOptions:
     value:           CamelCase
   - key:             llvm-namespace-comment.ShortNamespaceLines
     value:           '25'
-
-- 
2.20.1

